### PR TITLE
Keep track of internally updated selectedKey in Nav, add keys for Nav items in AppState example pages

### PR DESCRIPTION
**Pull request by cmalonzo**
Same PR, just had some fast-forwarding mishaps that needed a fresh new PR. Comments were addressed from the last one.

https://github.com/OfficeDev/office-ui-fabric-react/pull/290

### CODE REVIEW

File: src/components/Nav/Nav.tsx [hunk 1]
```diff
@@ -196,6 +194,14 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
     ev.preventDefault();
     ev.stopPropagation();
   }
+
+  private _onNavLinkClicked(link: INavLink, ev: React.MouseEvent): void {
+    this._selectedKey = link.key;
+
+    if (link.onClick) {
+      link.onClick(ev);
```
Comment (antonlabunets) [inline src/components/Nav/Nav.tsx]: Make sure to test with both button/href cases to ensure unnecessary navigation doesn't lead away the page. ie. preventPropagation etc. But maybe the design is that the click hadler is expected to do that?

Comment (dzearing) [inline src/components/Nav/Nav.tsx]: you man want to pass in the link as the 2nd param. The onClick definition could be:

onClick: (ev?: React.MouseEvent, link?: INavLink)

Comment (cmalonzo) [inline src/components/Nav/Nav.tsx]: Would this mean passing the link back to itself?

Comment (dzearing) [inline src/components/Nav/Nav.tsx]: that should be the caller's responsibility for sure. Consider a scenario like: have an href to navigate away, but have an onclick that logs telemetry. In that case, you want default behavior, but you still want the onclick to work.

Comment (cmalonzo) [inline src/components/Nav/Nav.tsx]: Just added this in and updated the props for INavLink's onClick

Comment (cmalonzo) [inline src/components/Nav/Nav.tsx]: Ah! I see the correct fix. The anchor link and the button link depend on different onClicks. Modified the logic to reflect this again.

File: src/components/Nav/Nav.tsx [hunk 2]
```diff
@@ -196,6 +194,22 @@ export class Nav extends React.Component<INavProps, INavState> implements INav {
...
+    if (this.props.onLinkClick) {
+      this.props.onLinkClick(ev, link);
+    }
+  }
+
+  private _onNavButtonLinkClicked(link: INavLink, ev: React.MouseEvent): void {
+    this._selectedKey = link.key;
+
+    if (link.onClick) {
+      link.onClick(ev);
```
Comment (dzearing) [inline src/components/Nav/Nav.tsx]: missing link here.

Comment (cmalonzo) [inline src/components/Nav/Nav.tsx]: Seemed there was a commit override that took out a commit that did this already. Changed!

**Comment by antonlabunets** [comment]
Approved the "old way".

File: src/components/Nav/Nav.tsx (hunk 2 above)
Comment (dzearing) [inline src/components/Nav/Nav.tsx]: Still not seeing link.onClick(ev, link) here on 210.

**Pull request merged**